presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Statistics.java

```diff
@@ -91,6 +91,12 @@ public static HiveBasicStatistics reduce(HiveBasicStatistics first, HiveBasicSta
 
     public static Map<String, HiveColumnStatistics> merge(Map<String, HiveColumnStatistics> first, Map<String, HiveColumnStatistics> second)
     {
+        // To correctly merge statistics during temporary table finish insertion. When ``first'' have exactly the same columns as ``second'' but all empty statistics,
+        // then the ``first'' is the placeholder empty statistics left at temporary table creation and safe to directly return second.
+        if (first.values().stream().allMatch(statistics -> statistics.equals(HiveColumnStatistics.empty()))
```

Comment:
@arhimondr can you check this logic?

Reply:
How about instead of adding this logic we set all statistics to 0 when creating a temporary table?

Reply:
You can use this helper method: https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Statistics.java#L244